File: data/derived_cot/rq3_prompting/results/translation/qwen/cpp_to_py/code_output/AutomaticGuitarSimulator.py
from typing import List

class PlayItem:
    def __init__(self, chord: str, tune: str):
        self.chord = chord
        self.tune = tune

class AutomaticGuitarSimulator:
    def __init__(self, text: str):
        self.play_text = text

    def interpret(self, display: bool = False) -> List[PlayItem]:
        if not self.play_text.strip():
            return []
        segments = self.play_text.split()
        play_list = []
        for seg in segments:
            i = 0
            while i < len(seg) and seg[i].isalpha():
                i += 1
            chord = seg[:i]
            tune = seg[i:]
            item = PlayItem(chord, tune)
            play_list.append(item)
            if display:
                print(self.format_display(chord, tune), flush=True)
        return play_list

    def format_display(self, key: str, value: str) -> str:
        return "Normal Guitar Playing -- Chord: " + key + ", Play Tune: " + value

File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/py_to_cpp/AutomaticGuitarSimulator.cpp
#include <vector>
#include <string>
#include <cctype>
#include <algorithm>
#include <sstream>

struct ChordTune {
    std::string chord;
    std::string tune;
};

class AutomaticGuitarSimulator {
private:
    std::string play_text;

    bool isAllWhitespace(const std::string& str) {
        if (str.empty()) return true;
        for (char c : str) {
            if (!std::isspace(static_cast<unsigned char>(c))) {
                return false;
            }
        }
        return true;
    }

public:
    explicit AutomaticGuitarSimulator(const std::string& text) : play_text(text) {}

    std::vector<ChordTune> interpret(bool display = false) {
        if (play_text.empty() || isAllWhitespace(play_text)) {
            return {};
        }

        std::vector<ChordTune> play_list;
        std::istringstream iss(play_text);
        std::string play_seg;

        while (std::getline(iss, play_seg, ' ')) {
            if (play_seg.empty()) {
                continue;
            }

            size_t pos = 0;
            for (size_t i = 0; i < play_seg.size(); ++i) {
                if (!std::isalpha(static_cast<unsigned char>(play_seg[i]))) {
                    pos = i;
                    break;
                }
            }

            std::string chord = play_seg.substr(0, pos);
            std::string tune = play_seg.substr(pos);

            play_list.push_back({chord, tune});

            if (display) {
                format_display(chord, tune);
            }
        }

        return play_list;
    }

    std::string format_display(const std::string& key, const std::string& value) {
        return "Normal Guitar Playing -- Chord: " + key + ", Play Tune: " + value;
    }
};